### PR TITLE
Undoing changes will scroll us back to the changed area in specific case

### PR DESCRIPTION
**Pull request by wszymanski**
### Context

As the title says. For now, it meets the requirements only from the case [described in the comment.](https://github.com/handsontable/dev-handsontable/issues/115#issuecomment-1833436440)

> 1. I edit a cell and press enter
> 2. I scroll down the grid and click on a new cell
> 3. I press ctrl z and my focus goes back to the initial cell I edited
> 4. The view should auto scroll with that cell in the top of the view

Please keep in mind that it moves vertically or horizontally only when there is such a need.

### How has this been tested?

Manual tests with scrolling on both axes.

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature or improvement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Additional language file or change to the existing one (translations)

### Related issue(s):
1. https://github.com/handsontable/dev-handsontable/issues/115

### Affected project(s):
- [x] `handsontable`
- [ ] `@handsontable/angular`
- [ ] `@handsontable/react`
- [ ] `@handsontable/vue`
- [ ] `@handsontable/vue3`

### Checklist:

- [x] I have reviewed the guidelines about [Contributing to Handsontable](https://github.com/handsontable/handsontable/blob/master/CONTRIBUTING.md) and I confirm that my code follows the code style of this project.
- [ ] My change requires a change to the documentation.

### CODE REVIEW

File: handsontable/src/plugins/undoRedo/undoRedo.js [hunk 1]
```diff
@@ -457,6 +457,37 @@ UndoRedo.ChangeAction.prototype.undo = function(instance, undoneCallback) {
...
+    const [selectedRow, selectedColumn] = selectedLast;
+    const firstFullyVisibleRow = instance.view.getFirstFullyVisibleRow();
+    const firstFullyVisibleColumn = instance.view.getFirstFullyVisibleColumn();
+    const isInVerticalViewPort = changedRow >= firstFullyVisibleRow;
+    const isInHorizontalViewPort = changedColumn >= firstFullyVisibleColumn;
+    const isInViewport = isInVerticalViewPort && isInHorizontalViewPort;
+    const isChangedSelection = selectedRow !== changedRow || selectedColumn !== changedColumn;
+
+    // Performing scroll only when selection has been changed right after editing a cell.
+    if (isInViewport === false && isChangedSelection === true) {
```
Comment (jansiegel) [inline handsontable/src/plugins/undoRedo/undoRedo.js:473]: I'm not sure about the `isChangedSelection` condition. 
It _is_ in line [with this comment](https://github.com/handsontable/dev-handsontable/issues/115#issuecomment-1833436440), but does not meet the requirements of the original post of https://github.com/handsontable/dev-handsontable/issues/115 (it does not require selecting a different cell before hitting `cmd+z`)

Is this what we want to do with it? 
cc: @evanSe

Comment (evanSe) [reply handsontable/src/plugins/undoRedo/undoRedo.js:473]: @jansiegel the current implementation in line with the comment is correct, @wszymanski we want a code sandbox for @krzysztofspilka review before this is merged to develop.

Comment (wszymanski) [reply handsontable/src/plugins/undoRedo/undoRedo.js:473]: > @jansiegel the current implementation in line with the comment is correct, @wszymanski we want a code sandbox for @krzysztofspilka review before this is merged to develop.

Here you go: https://jsfiddle.net/916z0tvc

Comment (evanSe) [reply handsontable/src/plugins/undoRedo/undoRedo.js:505]: @wszymanski we are good to merge

**Comment by wszymanski** [comment]
@jansiegel I need your ✅

**Review by jansiegel — APPROVED** [review]

**Pull request merged**